Common/Securities/SecurityTransactionManager.cs

```diff
@@ -296,12 +400,25 @@ public void WaitForOrder(int orderId)
         /// <returns>List of open orders.</returns>
         public List<Order> GetOpenOrders()
         {
-            var openOrders = (from order in Orders.Values
+            var openOrders = (from order in _orders.Values
                 where (order.Status == OrderStatus.Submitted ||
                        order.Status == OrderStatus.New)
                 select order).ToList();
 
             return openOrders;
+        }
+
+        /// <summary>
+        /// Get a list of all open orders.
+        /// </summary>
+        /// <returns>List of open orders.</returns>
+        public List<Order> GetOrders(Predicate<Order> filter = null)
+        {
+            var result = (from order in _orders.Values
```

Comment:
Calling `Values` requires all locks within the collection, the preferred way is to just enumerate `_orders` as `KeyValuePairs` and then to access the individual `Value` properties on each `KeyValuePair`